sgx-types/src/ssa.rs

```diff
@@ -270,45 +272,143 @@ pub struct Miscellaneous {
     exinfo: ExceptionInfo,
 }
 
-/// The size of XSAVE region in SSA is derived from the enclave’s support of the collection
-/// of processor extended states that would be managed by XSAVE. The enablement of those
-/// processor extended state components in conjunction with CPUID leaf 0DH information
-/// determines the XSAVE region size in SSA.
-///
-/// Section 38.9, Table 38-7
+/// This type represents an MMX register, used to perform operations on 64-bit packed integer data.
+/// It includes the 10-byte value as well as 6 bytes of padding.
+#[derive(Debug, Default)]
+#[repr(C)]
+pub struct Mm(u128);
+
+/// This type represents an XMM 128-bit data register, used to operate on packed single-precision floating-point operands.
+#[derive(Debug, Default)]
+#[repr(C)]
+pub struct Xmm(u128);
+
+// TODO: From implementations to u64 that mask off the relevant bits.
+
+/// The legacy region of an XSAVE area comprises the 512 bytes starting at the area’s base address.
+/// See Table 13-1 in
+/// https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-1-manual.pdf
 #[derive(Debug)]
-#[repr(C, align(4096))]
-pub struct XSave([[u64; 32]; 16]);
+#[repr(C)]
+pub struct XSaveLegacy {
+    /// x87 Floating Point Unit (FPU) Control Word
+    pub fcw: u16,
+    /// x87 FPU Status Word
+    pub fsw: u16,
+    /// x87 FPU Tag Word
+    pub ftw: u8,
+    reserved0: u8,
+    /// x87 FPU Opcode
+    pub fop: u16,
+    /// x87 FPU Instruction Pointer Offset
+    pub fip: u64,
+    /// x87 FPU Data Pointer Offset
+    pub fdp: u64,
+    /// 32-bit register providing status and control bits used in SIMD floating-point operations
+    pub mxcsr: Masked<u32>,
+    /// Register used to perform operations on 64-bit packed integer data
+    pub mm: [Mm; 8],
+    /// 128-bit data register used to operate on packed single-precision floating-point operands
+    pub xmm: [Xmm; 16],
+    // Size of XSaveLegacy must be 512.
+    padding0: [u64; 11],
+    padding1: [u8; 7],
+}
 
-/// The Footer combines the padding, Miscellaneous, and GPRSGX fields of the SSA Frame
-/// struct. It has no direct equivalent in the SGX documentation, but creates an SSA Frame
-/// in combination with the XSave struct.
-///
-/// Section 38.9, Table 38-7
+impl Default for XSaveLegacy {
+    fn default() -> Self {
```

Comment:
It might be shorter to use `MaybeUninit::zeroed()` here and then fill in the fields you want.